src/main/java/org/mule/extension/socket/api/connection/tcp/protocol/StreamingProtocol.java

```diff
@@ -50,20 +47,11 @@ public InputStream read(InputStream is) throws IOException {
 
 
   @Override
-  public void write(OutputStream os, Object data, String encoding) throws IOException {
-    if (data instanceof CursorStreamProvider) {
-      data = ((CursorStreamProvider) data).openCursor();
-    }
-
+  public void write(OutputStream os, InputStream data) throws IOException {
     try {
-      if (data instanceof InputStream) {
-        InputStream is = (InputStream) data;
-        copyLarge(is, os);
-        is.close();
-      } else {
-        this.writeByteArray(os, getByteArray(data, true, encoding, objectSerializer));
-      }
+      copyLarge(data, os);
     } finally {
+      data.close();
```

Comment:
this can throw exception and then the two lines below would never be executed. Use IOUtils.closeQuietly